ext/mruby_engine/mruby_engine.c

```diff
@@ -209,10 +210,22 @@ static struct RProc *generate_code(
   const char *source,
   me_host_exception_t *err)
 {
+  struct mrb_jmpbuf *previous = state->jmp;
+  struct mrb_jmpbuf p_jmp;
+  state->jmp = &p_jmp;
   struct mrb_parser_state *parser_state = mrb_parser_new(state);
   parser_state->s = source;
   parser_state->send = source + strlen(source);
-  mrb_parser_parse(parser_state, context);
+
+  MRB_TRY(state->jmp) {
+    mrb_parser_parse(parser_state, context);
+  }
+  MRB_CATCH(state->jmp) {
+    parser_state->nerr = 1;
```

Comment:
It is... the `parser_state` is fine... And that way we get the error as we already got to it